packages/terra-search-field/src/SearchField.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconSearch from 'terra-icon/lib/icon/IconSearch';
+import classNames from 'classnames';
+import './SearchField.scss';
+
+const propTypes = {
+  /**
+   * Placeholder text to show while the search field is empty.
+   */
+  placeholder: PropTypes.string,
+
+  /**
+   * The minimum number of characters to perform a search.
+   */
+  minimumSearchTextLength: PropTypes.number,
+
+  /**
+   * How long the component should wait after input before performing an automatic search.
+   */
+  searchDelay: PropTypes.number,
+
+  /**
+   * A callback to perform search. Sends parameter {String} searchText.
+   */
+  onSearch: PropTypes.func,
+};
+
+const defaultProps = {
+  placeholder: '',
+  minimumSearchTextLength: 2,
+  searchDelay: 250,
+};
+
+class SearchField extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.handleTextChange = this.handleTextChange.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+
+    this.searchTimeout = null;
+    this.state = {
+      searchText: '',
+    };
+  }
+
+  handleTextChange(event) {
+    this.setState({ searchText: event.target.value });
+
+    if (!this.searchTimeout) {
+      this.searchTimeout = setTimeout(this.handleSearch, this.props.searchDelay);
```

Comment:
I ran into a few issues with timeouts and un mounting where a timeout would be initialized but the component unmounted before the callback was fired and it would throw some errors.

It might be worth it to clear this timeout when the component unmounts